index.js

```diff
@@ -135,6 +128,20 @@ class ServerlessDynamodbLocal {
         return host;
     }
 
+    /**
+     * To check if the handler needs to be executed based on stage
+     *
+     * @return {Boolean} if the handler can run for the provided stage
+     */
+    shouldExecute() {
+      const stage = (this.options && this.options.stage) || (this.service.provider && this.service.provider.stage);
+      if (this.config.stages && !this.config.stages.includes(stage)) {
+        this.serverlessLog("DynamoDB Local is not available for stage: " + stage);
```

Comment:
:bulb: because of the way this check is utilized, anyone running this could end up seeing this exact log message repeated several times. A cleaner solution might be to add specific `serverlessLog`s to  each lifecycle step and explicitly call out why _that specific_ step didn't run. eg:

```js
migrateHandler() {
    if (this.shouldExecute()) {
        // ...do migration
    } else {
        this.serverlessLog("Skipping migration: DynamoDB Local is not available for stage: " + this.stage);
    }
}
```

**Note:** The specific example above would require extracting the `stage` property into a getter. That's up to you if you feel that the current `stage` _must_ be logged.

Reply:
I agree @JordanForeman I have updated the code as per your suggestion. Also, `install` and `remove` handler will not go under check, as this can be done for any `stage` in local.

Thanks